zebROS_ws/src/behaviors/src/2023_updown_elevater_server.cpp

```diff
@@ -0,0 +1,214 @@
+#include <ros/ros.h>
+#include <actionlib/server/simple_action_server.h>
+#include <behavior_actions/Elevater2023Action.h>
+#include <ddynamic_reconfigure/ddynamic_reconfigure.h>
+#include <std_msgs/Float64.h>
+#include <controllers_2023_msgs/ElevatorSrv.h>
+#include <iostream>
+
+#define ElevaterINFO(x) ROS_INFO_STREAM("2023_elevater_server : " << x)
+#define ElevaterERR(x) ROS_ERROR_STREAM("2023_elevater_server : " << x)
+
+typedef behavior_actions::Elevater2023Goal elevater_ns;
+/*
+uint8 INTAKE=0
+uint8 LOW_NODE=1
+uint8 MIDDLE_NODE=2
+uint8 HIGH_NODE=3
+uint8 mode
+
+uint8 CUBE=0
+uint8 VERTICAL_CONE=1
+uint8 BASE_TOWARDS_US_CONE=2 
+uint8 BASE_AWAY_US_CONE=3
+uint8 piece
+*/
+constexpr std::array mode_to_string = {"INTAKE", "LOW_NODE", "MIDDLE_NODE", "HIGH_NODE"};
+constexpr std::array piece_to_string = {"CUBE", "VERTICAL_CONE", "BASE_TOWARDS_US_CONE", "BASE_AWAY_US_CONE"};
+
+template <class T>
+void load_param_helper(const ros::NodeHandle &nh, std::string name, T &result, T default_val) {
+    if (!nh.getParam(name, result))
+    {
+      ElevaterERR("Could not find" << name << ", defaulting to " << default_val);
+      result = default_val;
+    }
+}
+
+class ElevaterAction2023 {
+
+protected:
+
+  ros::NodeHandle _nh;
+  ros::NodeHandle _nh_params;
+  actionlib::SimpleActionServer<behavior_actions::Elevater2023Action> _as;
```

Comment:
Don't start variables with an underscore.  Put the _ after the variable name if you want to mark it as a member var.